contracts/DAOTreasury.cdc

```diff
@@ -143,9 +143,33 @@ pub contract DAOTreasury {
     }
 
     destroy() {
-      destroy self.multiSignManager
-      destroy self.vaults
-      destroy self.collections
+      // Check if Valuts are empty
+      var isVaultEmpty = true
+      for identifier in self.vaults.keys {
+        let vaultRef = (&self.vaults[identifier] as &FungibleToken.Vault?)!
+        if vaultRef.balance > 0.0 {
```

Comment:
Another approach would be to write it like this:
```
assert(vaultRef.balance > 0.0, message: "Vault " + indentifier + " is not empty!")
```
In this case, the code will fail faster, with a message pointing to the vault that is not empty, instead of going through all the code in destroy